sample/TKTransferSamples.m

```diff
@@ -0,0 +1,215 @@
+    
+    __block int waitingForPayment = false;
+    
+    // notifyPaymentRequest begin snippet to include in docs
+    TokenPayload *payload = [TokenPayload message]; // hoped-for payment
+    payload.description_p = @"lunch";
+    payload.from.alias = payerAlias;
+    payload.to.id_p = payee.id;
+    payload.transfer.lifetimeAmount = @"100";
+    payload.transfer.currency = @"EUR";
+    
+    [tokenIO notifyPaymentRequest:payload
+                        onSuccess:^ {
+                            // Notification sent.
+                            waitingForPayment = true;
+                        } onError:^(NSError *e){
+                            // Something went wrong.
+                            // Maybe we used wrong alias.
+                            // Check error.
+                        }
+     ];
+     // notifyPaymentRequest done snippet to include in docs
+    
+    [self runUntilTrue:^ {
+        return waitingForPayment;
+    }];
+}
+
+- (void)testCreateEndorseTransferToken {
+    TokenIOSync *tokenIOSync = [[self sdkBuilder] buildSync];
+    TKMemberSync *payerSync = [self createMember:tokenIOSync];
+    TKMember *payer = payerSync.async;
+    TKAccountSync *payerAccountSync = [payerSync linkAccounts:[self createBankAuthorization:payerSync]][0];
+    TKAccount *payerAccount = payerAccountSync.async;
+    Alias *payeeAlias = [self generateAlias];
+    TKMemberSync *payeeSync = [tokenIOSync createMember:payeeAlias];
+    TKMember *payee = payeeSync.async;
+    NSString *refId = @"purchase:2017-11-01:28293336394ffby";
+    __block Token *transferToken = nil;
+    
+    // createTransferToken begin snippet to include in docs
+    TransferTokenBuilder *builder = [payer createTransferToken:100.0
+                                                      currency:@"EUR"];
+    builder.accountId = payerAccount.id;
+    builder.redeemerAlias = payeeAlias;
+    builder.descr = @"Book purchase";
+    builder.refId = refId;
+    
+    [builder executeAsync:^(Token *t){
+        // Use token.
+        transferToken = t;
+    }   onError:^(NSError *e) {
+        // Something went wrong. (We don't just build a structure; we also
+        // upload it to Token cloud. So things can go wrong.)
+        @throw [NSException exceptionWithName:@"BuilderExecuteException"
+                                       reason:[e localizedFailureReason]
+                                     userInfo:[e userInfo]];
+    }];
+    // createTransferToken done snippet to include in docs
+    
+    [self runUntilTrue:^ {
+        return (transferToken != nil);
+    }];
+    
+    // endorseToken begin snippet to include in docs
+    [payer endorseToken:transferToken
+                withKey:Key_Level_Standard
+              onSuccess:^(TokenOperationResult *result) {
+                  // Update transferToken with newer value:
+                  // Payload is same; now has signatures attached.
+                  transferToken = result.token;
```

Comment:
Maybe you would like to check the signature here. If the backend have a bug return a token without signature, the thread will be hung. No sure if we need to handle this.

Reply:
I couldn't think of a good reason to do the check that made a realistic sample.
The "runUntilTrue" times out after a few seconds, so thread won't hang forever.

If you have a reason to check, I'll use it.

Reply:
ok